TLM/TLM/Custom/PathFinding/CustomPathFind.cs

```diff
@@ -289,8 +291,16 @@ private struct BufferItem {
                 disableMask_ |= NetSegment.Flags.Flooded;
             }
 
+            allowMixedCargoPath_ = false;
+            mixedPathVehicleTypes_ = queueItem_.vehicleType;
             if ((laneTypes_ & NetInfo.LaneType.Vehicle) != NetInfo.LaneType.None) {
                 laneTypes_ |= NetInfo.LaneType.TransportVehicle;
+
+                // allowed mixed vehicle path when requeted path is Vehicle + CargoVehicle (at current state CargoTruck and PostVanAI)
+                allowMixedCargoPath_ = (laneTypes_ & NetInfo.LaneType.CargoVehicle) != NetInfo.LaneType.None;
+                if (allowMixedCargoPath_) {
+                    mixedPathVehicleTypes_ = VehicleTypeToMixedCargoExtVehicle(vehicleTypes_, queueItem_.vehicleType);
```

Comment:
It's like that only because I found another bug in the code recently.

Reply:
so it was a oversight huh? are you going to change it?

Reply:
so it was an oversight? are you going to fix this?
or are you trying my suggestion is somehow a bug?

Reply:
@kianzarrin I was meant the code was written like that and made sense, but only before I found a bug which changes a lot of things, and now.. it can be simplified as you've pointed out.